helm-charts/splunk-otel-collector/templates/configmap-fluentd.yaml

```diff
@@ -216,49 +210,55 @@ data:
         <record>
           # set the sourcetype from splunk.com/sourcetype pod annotation or set it to kube:container:CONTAINER_NAME
           sourcetype ${record.dig("kubernetes", "annotations", "splunk.com/sourcetype") ? "kube:"+record.dig("kubernetes", "annotations", "splunk.com/sourcetype") : "kube:container:"+record.dig("kubernetes","container_name")}
-          container_name ${record.dig("kubernetes","container_name")}
-          namespace ${record.dig("kubernetes","namespace_name")}
-          pod ${record.dig("kubernetes","pod_name")}
-          container_id ${record.dig("docker","container_id")}
-          pod_uid ${record.dig("kubernetes","pod_id")}
-          node_name "#{ENV['K8S_NODE_NAME']}"
-          cluster_name {{ .Values.clusterName }}
-          container_image ${record.dig("kubernetes","container_image")}
           # set the index field to the value found in the pod splunk.com/index annotations. if not set, use namespace annotation, or default to the default_index
-          index ${record.dig("kubernetes", "annotations", "splunk.com/index") ? record.dig("kubernetes", "annotations", "splunk.com/index") : record.dig("kubernetes", "namespace_annotations", "splunk.com/index") ? (record["kubernetes"]["namespace_annotations"]["splunk.com/index"]) : ("{{ .Values.logsBackend.hec.indexName | default "main"}}")}
+          index ${record.dig("kubernetes", "annotations", "splunk.com/index") ? record.dig("kubernetes", "annotations", "splunk.com/index") : record.dig("kubernetes", "namespace_annotations", "splunk.com/index")}
```

Comment:
Look like `main` default value is missed?

Reply:
I tried to consolidate setting of index default in the filter ** section:

```
com.splunk.index ${record.dig("index") ? record.dig("index") : "{{ .Values.logsBackend.hec.indexName | default "main"}}"}
```

I'll have to test more to be completely confident it's working.

Reply:
Yeah seems to work.

Reply:
So this will be updated, right?

Reply:
If index isn't set or is null then it falls back to setting default value here. No need to set default everywhere.

Reply:
Ok found it in `com.splunk.index`